### PR TITLE
3.0 - Orders - Update Tax Rate When Selecting Existing Address

### LINKED ISSUE #7958: 3.0 - Selecting an existing address for a manual order does not retrieve the region
### Expected behavior
When I manually create an order and select a customer's preexisting address, if it is within a region which collects taxes, I expect the taxes to be applied.

### Actual behavior
When I select the address, the `getTaxRate` JS function retrieves the country, but not the region/state, so the tax rate is not applied.

### Steps to reproduce the behavior
1) In an EDD store with taxes enabled for a region (not country wide), create a new order.
2) The order should be for an existing customer with an existing address which meets the criteria for being taxable (so I picked a customer with a TN address).
3) Add a product to the order. It will show a 0.00% tax rate.
4) Steps 2 and 3 can be done in reverse order; it doesn't seem to change it.

### Information (if a specific version is affected):
PHP Version: 7.3.5
EDD Version (or branch): 3.0
WordPress Version: 5.4.1

Any other relevant information:
I thought we had this resolved at one point ([ref](https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7781#discussion_r414800661)), but maybe that applied to addresses being typed in, not retrieving an existing address.

### PR DESCRIPTION
**Pull request by spencerfinnell**
Fixes #7958 

Proposed Changes:
1. Ensure tax rate is updated correctly when selecting an existing address.
2. Allow address to be cleared by selecting blank existing address.

### CODE REVIEW

File: includes/ajax-functions.php [hunk 1]
```diff
@@ -1227,8 +1233,8 @@ function edd_ajax_customer_addresses() {
 
 			// Fetch the select
 			if ( ! empty( $options ) ) {
-				$html  = '<select data-nonce="' . wp_create_nonce( 'edd-country-field-nonce' ) . '" data-placeholder="Select a previously used address" class="edd-select-chosen ' . esc_attr( $rtl_class ) . ' add-order-customer-address-select">';
-				$html .= '<option></option>';
+				$html  = '<select id="edd_customer_existing_addresses" data-nonce="' . wp_create_nonce( 'edd-country-field-nonce' ) . '" data-placeholder="Select a previously used address" class="edd-select-chosen ' . esc_attr( $rtl_class ) . ' add-order-customer-address-select">';
```
Comment (robincornett) [inline includes/ajax-functions.php:1236]: I don't think the `edd-select-chosen` class is necessary here--that script is not implemented or applied.

**Comment by robincornett** [comment]
@spencerfinnell I just realized that on smaller screens, the `#edd_customer_existing_addresses` select overflows the bounds of the container. It looks like the core `select` CSS sets a max-width, but not a width, which may be the issue here.

**Comment by spencerfinnell** [comment]
@robincornett I added a `max-width` rule for `select.edd-form-group__input` in https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7973/commits/65eeb93b0b02c49721e6b22c25676f4bcf73a18f

**Comment by robincornett** [comment]
Yes, that fixes it, thanks!

File: includes/ajax-functions.php [hunk 2]
```diff
@@ -538,9 +538,15 @@ function edd_ajax_get_states_field() {
 	// Maybe setup the new listbox.
 	if ( ! empty( $states ) ) {
 		$field_name = sanitize_text_field( $_POST['field_name'] );
+		$field_id   = sanitize_text_field( $_POST['field_id'] );
```
Comment (ashleyfae) [inline includes/ajax-functions.php:541]: This causes the following notice when adding a tax rate in settings

```
PHP Notice:  Undefined index: field_id in /var/www/edd3/wp-content/plugins/easy-digital-downloads/includes/ajax-functions.php on line 541
```

1. Go to Downloads > Settings > Taxes.
2. Select a country via the dropdown.
3. Notice triggers when states are loaded for that country.

Comment (spencerfinnell) [reply includes/ajax-functions.php]: @ashleyfae Updated. I also added a check and fallback for `field_name` which should never be empty but could be with incorrect implementation.

**Comment by ashleyfae** [comment]
Someone help me... what am I doing wrong.. lol

## Tax rate:

Screenshot from 2020-06-03 14-37-24

## Order address

Screenshot from 2020-06-03 14-38-01

## Order tax (incorrect)

Screenshot from 2020-06-03 14-38-28

Should be 20% tax.

**Comment by spencerfinnell** [comment]
@ashleyfae Were you stung by https://github.com/easydigitaldownloads/easy-digital-downloads/issues/7629 -- the rate should say "Deactivate" -- it stays on "Removed" until settings are saved.

**Comment by ashleyfae** [comment]
> @ashleyfae Were you stung by #7629 -- the rate should say "Deactivate" -- it stays on "Removed" until settings are saved.

sdfsiokdsjfoswpfijoifwj

You're absolutely right. FML.

After _actually saving_ this works perfectly.

**Pull request merged**